tests/rewriting/test_html_rewriting.py

```diff
@@ -1308,11 +1329,16 @@ def test_html_tag_rewrite_rules(
     *,
     auto_close: bool,
     expected_result: str | None,
+    simple_url_rewriter_gen: Callable[[str, str], ArticleUrlRewriter],
 ):
     assert (
         rules.do_tag_rewrite(
             tag=tag,
             attrs=attrs,
+            url_rewriter=simple_url_rewriter_gen(
+                "http://www.example.com", "?queryparam"
+            ),
+            base_href=None,
```

Comment:
No test with an actual `base_href`?

Reply:
Indeed, just added one. Codecov does not complain because it is covered by `base_href = None`, but this does not really check we really pass proper value up to the rule.